pkg/git/libgit2/managed/transport_test.go

```diff
@@ -25,6 +36,38 @@ func TestKnownHostsCallback(t *testing.T) {
 		hostkey      git2go.HostkeyCertificate
 		want         error
 	}{
+		{
+			name:         "Empty",
+			host:         "source.developers.google.com",
+			knownHosts:   []byte(""),
+			hostkey:      git2go.HostkeyCertificate{Kind: git2go.HostkeySHA256, HashSHA256: sha256Fingerprint("AGvEpqYNMqsRNIviwyk4J4HM0lEylomDBKOWZsBn434")},
+			expectedHost: "source.developers.google.com:2022",
+			want:         fmt.Errorf("hostkey verification aborted: no known_hosts found"),
+		},
+		{
+			name:         "Mismatch incorrect known_hosts",
+			host:         "source.developers.google.com",
+			knownHosts:   []byte(knownHostsFixtureUnormalized),
+			hostkey:      git2go.HostkeyCertificate{Kind: git2go.HostkeySHA256, HashSHA256: sha256Fingerprint("AGvEpqYNMqsRNIviwyk4J4HM0lEylomDBKOWZsBn434")},
+			expectedHost: "source.developers.google.com:2022",
+			want:         fmt.Errorf("no entries in known_hosts matches host '[source.developers.google.com]:2022' with fingerprint 'AGvEpqYNMqsRNIviwyk4J4HM0lEylomDBKOWZsBn434='"),
```

Comment:
Pointing this out just to highlight how this looks like when printing the fingerprint in the error, so that we can discuss and think if there's a better way to express this error.

Reply:
Small point here, based on @aryan9600's suggestion, we will no longer add padding to the fingerprints.